plugin.js

```diff
@@ -38,9 +49,14 @@ function fastifyCookieOnReqHandler (fastifyReq, fastifyRes, done) {
 }
 
 function plugin (fastify, options, next) {
+  if (options.secret) {
+    fastify.decorateReply(cookieSignatureSecret, options.secret)
+  }
```

Comment:
Do we need to expose this? Why it's exposed on the Reply and not on the fastify object?

Reply:
I suggested it be attached to the instance instead of as a top level `let`.

Reply:
In the fastify object, performance increases, I think, this avoids adding a property to each request. But how i can add to the fastify object correctly, and it's a good practice?

Reply:
In the fastify object, performance increases, I think, this avoids adding a property to each request. I have a doubt about fastify decorators, if i do fastify.decorate ('prop', 'value'); it's accessible on the server instance, how can i access it from setCookie function?

Reply:
You should move the `fastifyCookieSetCookie` and `fastifyCookieUnsignCookie` functions within the `plugin` function itself, and define a variable in there that holds the secret.

Reply:
will cause break changes?

Reply:
no

Reply:
Well, i think this is not needed, just bind the setCookieFunction function passing a extra var with the secret.

like this:
fastify.decorateReply('setCookie', function setCookieWrapper(name, value, cookieOptions){
  return fastifyCookieSetCookie.call(this, name, value, cookieOptions, pluginOpitions.secret);
});

agre?

Reply:
I would prefer to avoid the usage of `call`, as it has a tiny bit of overhead.